mission_control/navigator_missions/nav_missions/scan_the_code.py

```diff
@@ -1,32 +1,60 @@
 #!/usr/bin/env python
+"""Scan The Code Mission Script."""
 import txros
 import numpy as np
-from navigator_msgs.srv import ObjectDBSingleQuery, ObjectDBSingleQueryRequest
+from navigator_scan_the_code import ScanTheCodeMission
+from geometry_msgs.msg import PoseStamped
+import navigator_tools as nt
+___author___ = "Tess Bianchi"
+
+
+def _publish_pose(pub, pose):
+    pose_stamped = PoseStamped()
+    pose_stamped.header.frame_id = "enu"
+    pose = nt.numpy_quat_pair_to_pose(pose.pose[0], pose.pose[1])
+    pose_stamped.pose = pose
+    pub.publish(pose_stamped)
 
 
 @txros.util.cancellableInlineCallbacks
 def main(navigator):
+    """Main Script of Scan The Code."""
     navigator.change_wrench("autonomous")
-    serv = navigator.nh.get_service_client("/database/single", ObjectDBSingleQuery)
-    req = ObjectDBSingleQueryRequest()
-    req.name = 'scan_the_code'
-    ans = yield serv(req)
-    print ans
-    if not ans.found:
-        return
-
-    obj = ans.object
-
-    stc_position = np.array([obj.position.x, obj.position.y, obj.position.z])
-
-    diff = np.subtract(stc_position, navigator.move.position)
-    diff = np.array([diff[0], diff[1], 0])
-    length = np.linalg.norm(diff)
-    radius = 4
-    if(length - radius >= 0):
-        d = navigator.move.rel_position(diff * (length - radius) / (length)).look_at(stc_position).go()
-        yield d
-    yield navigator.vision_request("scan_the_code_activate")
-    pattern = navigator.move.circle_point(stc_position, radius + 2, granularity=8)
-    searcher = navigator.search(vision_proxy='/vision/scan_the_code_status', search_pattern=pattern)
-    yield searcher.start_search(spotings_req=1, speed=.9)
+    pub = yield navigator.nh.advertise("/stc/pose", PoseStamped)
+    mission = ScanTheCodeMission(navigator.nh)
+    yield mission.init_(navigator.tf_listener)
+    pose, look_at = yield mission.initial_position()
+    initial_pose = navigator.move.set_position(pose).look_at(look_at)
+    yield navigator.nh.sleep(1)
+    _publish_pose(pub, initial_pose)
+
+    # UNCOMMENT
+    # yield navigator.move.set_position(pose).look_at(look_at).go()
+
+    mission.correct_pose(pose)
+    navigator.nh.sleep(.2)
+    if not mission.stc_correct:
+        circle = navigator.move.circle_point(look_at, 8, granularity=10)
+        for p in circle:
+            if mission.stc_correct:
+                break
+            # yield p.go()
+
+    colors = yield mission.find_colors()
+    print colors
+    if colors is None:
```

Comment:
So colors is None when the perception fails? Perhaps it should throw an exception / tell the mission planner it has failed or at least log it. 

Also, does this set any params / output for what it saw besides setting the detect deliver params? I think for the judge GUI we'll want params set for what scan the code saw (#118 will have a convenient way to do this)